lib/widgets/filterslist.dart

```diff
@@ -1,69 +1,76 @@
-        final isSelected = selectedFilters.contains(statusFilter[index]);
-        return FilterChip(
-          onSelected: (bool selected) => setState(
-            () {
-              selected
-                  ? selectedFilters.add(statusFilter[index])
-                  : selectedFilters.remove(statusFilter[index]);
-            },
-          ),
-          selected: isSelected,
-          label: Padding(
-            padding: const EdgeInsets.symmetric(horizontal: 7.0, vertical: 6.0),
-            child: Text(
-              statusFilter[index],
-              style: TextStyle(
-                color: isSelected
-                    ? const Color.fromRGBO(255, 255, 255, 1)
-                    : const Color.fromRGBO(188, 196, 220, 1),
-                fontSize: 20,
-                fontWeight: FontWeight.w300,
+      child: Padding(
+          padding: const EdgeInsets.fromLTRB(15, 0, 0, 0),
+          child: Row(
+            children: [
+              Wrap(
+                spacing: 8.0,
+                children: statusFilter.map((filter) {
+                  final isSelected = selectedFilters.contains(filter);
+                  return FilterChip(
+                    label: Text(
+                      filter,
+                      style: TextStyle(
+                        color: isSelected
+                            ? const Color.fromRGBO(255, 255, 255, 1)
+                            : const Color.fromRGBO(188, 196, 220, 1),
+                        fontWeight: FontWeight.w300,
+                      ),
+                    ),
+                    backgroundColor: const Color(0x00f2f7ff),
+                    shape: StadiumBorder(
+                      side: BorderSide(
+                          color: isSelected
+                              ? const Color.fromRGBO(0, 255, 0, 1)
+                              : const Color.fromRGBO(188, 196, 220, 1),
+                          width: 1),
+                    ),
+                    selectedColor: const Color.fromRGBO(0, 186, 136, 1),
+                    elevation: isSelected ? 4 : 0,
+                    selectedShadowColor: const Color.fromRGBO(0, 186, 136, 0.3),
+                    showCheckmark: false,
+                    selected: isSelected,
+                    onSelected: (selected) {
+                      if (selected && !selectedFilters.contains(filter)) {
+                        ref.read(usersProvider.notifier).addChip(filter);
+                        ref
+                            .read(selectedChipsProvider.notifier)
+                            .toggleChip(filter);
+                      } else if (selectedFilters.contains(filter)) {
+                        ref.read(usersProvider.notifier).removeChip(filter);
+                        ref
+                            .read(selectedChipsProvider.notifier)
+                            .toggleChip(filter);
+                      }
+                      debugPrint('$selectedFilters');
+                    },
```

Comment:
This is a product of bad state management. You currently have 2 providers that manage what chips are seleceted: _ usersProvider_ and _ selectedChipsProvider_.  _ selectedChipsProvider_ should be the one who handles what chips are selected.